server/routerlicious/packages/services-shared/src/startupChecker.ts

```diff
@@ -0,0 +1,32 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/**
+ * This class is used to check if a service/functionality has started up and is ready for use.
+ * This is a singleton class so that it need not be passed as a resource to all services where it is needed.
```

Comment:
Services launched through runner are running in separate execution environments, so having a singleton for sharing the state across services won't work in this case. My understanding is that the need for singleton is to be able to share the state between where it is being set (the runner) and where it is being checked (startup http endpoint handler)?

Reply:
Yes that is correct, this will specific to a process or an instance of a service. The startup singleton will not be shared b/w services. It will just be relevant to a single service and an instance of that service.